### DISCUSSION TITLE
[v2] Conditional capabilities vs. platform

### DISCUSSION THREAD

**Discussion by VPKSoft** [tauri-apps/tauri#10400] [Q&A] [answered]
I managed to make my application to not include window state plugin in case of android or ios and got it running on android emulator. The question is how can I make the permission files in the `src-tauri/capabilities` folder conditional as the build fails if I don't delete the the plugin-related permission file?
The related conditional code:
```rust
pub fn run() {
    #[cfg(any(target_os = "android", target_os = "ios"))]
    {
        tauri::Builder::default()
        .run(tauri::generate_context!())
         .expect("error while running tauri application");
    }

    #[cfg(not(any(target_os = "android", target_os = "ios")))]
    {
        tauri::Builder::default()
       .plugin(tauri_plugin_window_state::Builder::default().build());
       .run(tauri::generate_context!())
       .expect("error while running tauri application");
    }
}
```

And `Cargo.toml`:
```toml
[target."cfg(not(any(target_os = \"android\", target_os = \"ios\")))".dependencies]
tauri-plugin-window-state = "=2.0.0-beta.11"
```

But here is the error if the permission file is not deleted:
```
Permission window-state:allow-restore-state not found
...
warning: build failed, waiting for other jobs to finish...
    Error `Failed to run `cargo build`: command ["cargo", "build", ...
```

**Comment by FabianLars** [comment, marked as answer]
You can have multiple capabilities files like this for example https://github.com/tauri-apps/plugins-workspace/tree/v2/examples/api/src-tauri/capabilities

**Reply by VPKSoft** [reply]
Thanks, just what I was looking for 👍

**Reply by setoelkahfi** [reply]
@FabianLars This is not the same as conditionally initializing a plugin based on the platform, right? Or does it mean there's no need for plugin initialization? Sounds wrong to me. 

For example, I want to only initialize the global shortcut plugin for desktop, for obvious reason. And this code doesn't work:

```
#[cfg_attr(mobile, tauri::mobile_entry_point)]
pub fn run() {
    #[cfg(any(target_os = "android", target_os = "ios"))]
    {
        run_mobile();
    }
    #[cfg(not(any(target_os = "android", target_os = "ios")))]
    {
        run_desktop();
    }
}
```

Am I missing something?

**Reply by FabianLars** [reply]
> This is not the same as conditionally initializing a plugin based on the platform, right?

No, this is just platform specific permissions which you could use for plugins you enable on all platforms as well.

> Am I missing something?

In what way does it not work? Any errors? Without any context this code looks fine. The way we typically use platform specific plugins is more like this https://github.com/tauri-apps/plugins-workspace/blob/v2/examples/api/src-tauri/src/lib.rs#L43-L61 - typically the mobile and desktop setups aren't vastly different, or not enough to split into run_mobile and run_desktop for example, but this of course isn't always true and depends a bit on personal preference.

**Reply by setoelkahfi** [reply]
I found the issue. I should put the platform-specific plugin initialization in the setup closure, like the example you linked there.
